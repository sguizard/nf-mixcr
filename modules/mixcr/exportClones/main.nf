process MIXCR_EXPORTCLONES {
    tag "$meta.id"

    container 'ghcr.io/milaboratory/mixcr/mixcr:4.6.0'

    input:
    tuple val(meta), path(clns)
    path(library)

    output:
    tuple val(meta), path("*_exportClones_*.tsv"), emit: tables
    path  "versions.yml"                         , emit: versions

    script:
    def prefix = task.ext.prefix ?: meta.id

    """
    export _JAVA_OPTIONS="-XX:-UsePerfData"

    mixcr exportClones $clns ${prefix}_exportClones.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        MiXCR: \$( mixcr --version | grep -e 'MiXCR' | sed 's/MiXCR v//' | sed 's/ .\\+//' )
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: meta.id
    """
    touch ${prefix}_exportClones.tsv

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        MiXCR: \$( mixcr --version | grep -e 'MiXCR' | sed 's/MiXCR v//' | sed 's/ .\\+//' )
    END_VERSIONS
    """
}
