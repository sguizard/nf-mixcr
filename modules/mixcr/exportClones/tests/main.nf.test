nextflow_process {

    name "Test Process MIXCR_EXPORTCLONES"
    script "../main.nf"
    process "MIXCR_EXPORTCLONES"
    config "nextflow.config"

    tag "mixcr"
    tag "mixcr/exportClones"

    setup {
        run("MIXCR_ANALYZE") {
            script "../../analyze/main.nf"
            config "../../analyze/tests/nextflow.config"
            process {
                """
                input[0] = 
                [
                    ['id':'TCR_cat'],
                    [
                        file("${projectDir}/test-datasets/23570MA0002L01_1_20k.fastq.gz"),
                        file("${projectDir}/test-datasets/23570MA0002L01_2_20k.fastq.gz")
                    ]
                ]
                input[1] = 'generic-amplicon-with-umi'
                input[2] = file("${projectDir}/test-datasets/imgt.202312-3.sv8.json.gz")
                """
            }
        }
    }

    test("MIXCR_EXPORTCLONES Export clone tables") {

        when {
            process {
                """
                input[0] = MIXCR_ANALYZE.out.clns
                input[1] = file("${projectDir}/test-datasets/imgt.202312-3.sv8.json.gz")
                """
            }
        }

        then {
            assert process.success
            assert snapshot(process.out).match()
        }

    }

}
