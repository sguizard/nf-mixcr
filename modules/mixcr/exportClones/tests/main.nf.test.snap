{
    "MIXCR_EXPORTCLONES Export clone tables": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "TCR_cat"
                        },
                        [
                            "[id:TCR_cat]_exportClones_IGL.tsv:md5,dbb08d4d51c2ce9370e767aa40696e2e",
                            "[id:TCR_cat]_exportClones_TRB.tsv:md5,cd23934a15f41111c0e1bd580bfbed4d"
                        ]
                    ]
                ],
                "1": [
                    "versions.yml:md5,ecf1f51180f925c42ab6c6850c1f66a3"
                ],
                "tables": [
                    [
                        {
                            "id": "TCR_cat"
                        },
                        [
                            "[id:TCR_cat]_exportClones_IGL.tsv:md5,dbb08d4d51c2ce9370e767aa40696e2e",
                            "[id:TCR_cat]_exportClones_TRB.tsv:md5,cd23934a15f41111c0e1bd580bfbed4d"
                        ]
                    ]
                ],
                "versions": [
                    "versions.yml:md5,ecf1f51180f925c42ab6c6850c1f66a3"
                ]
            }
        ],
        "timestamp": "2024-01-22T12:05:23.456234"
    }
}